test/list/reduce.wren

```diff
@@ -0,0 +1,9 @@
+var a = [1, 4, 2, 1, 5]
+var max = new Fn {|a, b| a > b ? a : b }
+var sum = new Fn {|a, b| a + b }
```

Comment:
A few more things to test:

1. Test that it works on a range too.
2. Test an empty sequence causes a runtime error. (Put this in a separate test, please.)
3. It would be good to have a test that demonstrates the order that the fold happens. Maybe concatenate strings in the function?

I know it's a lot of tests (in addition to the great ones you already did), but stability is our lifeblood. :)